sonar-text-plugin/src/main/java/org/sonar/plugins/secrets/api/SpecificationLoader.java

```diff
@@ -63,17 +56,32 @@ private static Map<String, List<Rule>> initialize(String specificationLocation,
       try {
         specification = loadSpecification(specificationLocation, specificationFileName);
       } catch (DeserializationException | SchemaValidationException e) {
-        LOG.error(String.format("Could not load specification from file: %s", specificationFileName), e);
+        LOG.error(String.format("%s: Could not load specification from file: %s", e.getClass().getSimpleName(), specificationFileName));
         continue;
       }
 
       for (Rule rule : specification.getProvider().getRules()) {
-        keyToRule.computeIfAbsent(rule.getRspecKey(), k -> new ArrayList<>()).add(rule);
+        if (!ruleFunctionalityIsNotImplementedFor(rule)) {
+          keyToRule.computeIfAbsent(rule.getRspecKey(), k -> new ArrayList<>()).add(rule);
+        }
       }
     }
     return keyToRule;
   }
 
+  //TODO: SONARTEXT-44 Add the missing detection logic to the Specification based Check
+  private static boolean ruleFunctionalityIsNotImplementedFor(Rule rule) {
+    // TODO: SONARTEXT-31: Add Context to detection logic
+    Matching matching = rule.getDetection().getMatching();
+    boolean contextIsSet = matching != null && matching.getContext() != null;
+
+    // TODO: SONARTEXT-48: Extend Post-Filter to include heuristic Filter
+    PostModule post = rule.getDetection().getPost();
+    boolean heuristicFilterIsSet = post != null && post.getHeuristicFilter() != null;
+
+    return contextIsSet || heuristicFilterIsSet;
+  }
```

Comment:
I changed this behavior to only disable tests based on those two missing features. 
Pre-Filter and StatisticalFilter.inputString only server to disable FP's. IMO it doesn't make sense to remove functionality in between releases.